src/Rector/Closure/MinimalSharedStringSolver.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Symfony\Rector\Closure;
+
+/**
+ * Class Solver copied from https://github.com/Triun/PHP-Longest-Common-Substring
+ */
+final class MinimalSharedStringSolver
+{
+    /**
+     * @return string
+     */
+    public function solve(string $stringA, string $stringB)
+    {
+        if (func_num_args() > 2) {
+            $arguments = func_get_args();
+            array_splice($arguments, 0, 2, [$this->solve($stringA, $stringB)]);
+
+            return call_user_func_array([$this, 'solve'], $arguments);
+        }
+
+        $charsA = str_split($stringA);
+        $charsB = str_split($stringB);
+
+        $matrix = array_fill_keys(array_keys($charsA), array_fill_keys(array_keys($charsB), 0));
+        $longestLength = 0;
+        $longestIndexes = [];
+
+        foreach ($charsA as $i => $charA) {
+            foreach ($charsB as $j => $charB) {
+                if ($charA === $charB) {
+                    if ($i === 0 || $j === 0) {
+                        $matrix[$i][$j] = 1;
+                    } else {
+                        $matrix[$i][$j] = $matrix[$i - 1][$j - 1] + 1;
+                    }
+
+                    $newIndex = $this->newIndex($matrix, $i, $j);
+                    if ($matrix[$i][$j] > $longestLength) {
+                        $longestLength = $matrix[$i][$j];
+                        $longestIndexes = [$newIndex];
+                    } elseif ($matrix[$i][$j] === $longestLength) {
+                        $longestIndexes[] = $newIndex;
+                    }
+                } else {
+                    $matrix[$i][$j] = 0;
+                }
+            }
+        }
+
+        return $this->result($longestIndexes, $longestLength, $stringA);
+    }
+
+    /**
+     * @param array<int, array<int, int>> $matrix
+     *
+     * @return int
+     */
+    private function newIndex(array $matrix, int $i, int $j)
```

Comment:
add return type int here

Reply:
ahh didn't realize that I need to use native return types. maybe the error message should be more precise.

thanks - fixed.